x-pack/plugins/apm/public/components/app/service_inventory/service_inventory.test.tsx

```diff
@@ -5,249 +5,17 @@
-import {
-  mockApmPluginContextValue,
-  MockApmPluginContextWrapper,
-} from '../../../context/apm_plugin/mock_apm_plugin_context';
-import { FETCH_STATUS } from '../../../hooks/use_fetcher';
-import { clearCache } from '../../../services/rest/callApi';
-import * as useDynamicDataViewHooks from '../../../hooks/use_dynamic_data_view';
-import { SessionStorageMock } from '../../../services/__mocks__/SessionStorageMock';
-import { MockUrlParamsContextProvider } from '../../../context/url_params_context/mock_url_params_context_provider';
-import * as hook from '../../../context/anomaly_detection_jobs/use_anomaly_detection_jobs_context';
+import { composeStories } from '@storybook/testing-react';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import * as stories from './service_inventory.stories';
 
-const KibanaReactContext = createKibanaReactContext({
-  usageCollection: { reportUiCounter: () => {} },
-} as Partial<CoreStart>);
-
-const addWarning = jest.fn();
-const httpGet = jest.fn();
-
-function wrapper({ children }: { children?: ReactNode }) {
-  const mockPluginContext = merge({}, mockApmPluginContextValue, {
-    core: {
-      http: {
-        get: httpGet,
-      },
-      notifications: {
-        toasts: {
-          addWarning,
-        },
-      },
-    },
-  }) as unknown as ApmPluginContextValue;
-
-  return (
-    <MemoryRouter initialEntries={['/services?rangeFrom=now-15m&rangeTo=now']}>
-      <EuiThemeProvider>
-        <KibanaReactContext.Provider>
-          <MockApmPluginContextWrapper value={mockPluginContext}>
-            <MockUrlParamsContextProvider
-              params={{
-                rangeFrom: 'now-15m',
-                rangeTo: 'now',
-                start: '2021-02-12T13:20:43.344Z',
-                end: '2021-02-12T13:20:58.344Z',
-                comparisonEnabled: true,
-                comparisonType: TimeRangeComparisonEnum.DayBefore,
-              }}
-            >
-              {children}
-            </MockUrlParamsContextProvider>
-          </MockApmPluginContextWrapper>
-        </KibanaReactContext.Provider>
-      </EuiThemeProvider>
-    </MemoryRouter>
-  );
-}
+const { Example } = composeStories(stories);
 
 describe('ServiceInventory', () => {
-  beforeEach(() => {
-    // @ts-expect-error
-    global.sessionStorage = new SessionStorageMock();
-    clearCache();
-
-    jest.spyOn(hook, 'useAnomalyDetectionJobsContext').mockReturnValue({
-      anomalyDetectionJobsData: { jobs: [], hasLegacyJobs: false },
-      anomalyDetectionJobsStatus: FETCH_STATUS.SUCCESS,
-      anomalyDetectionJobsRefetch: () => {},
-    });
-
-    jest
-      .spyOn(useDynamicDataViewHooks, 'useDynamicDataViewFetcher')
-      .mockReturnValue({
-        dataView: undefined,
-        status: FETCH_STATUS.SUCCESS,
-      });
-  });
-
-  afterEach(() => {
-    jest.resetAllMocks();
-  });
-
-  it('should render services, when list is not empty', async () => {
```

Comment:
Are all these tests flaky?

Reply:
They were not, but they also were testing the things that we're already testing in https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/components/app/service_inventory/service_list/service_list.test.tsx.